test_session_tracing fails on machines with real session_trace config (Mem2MemConfig() reads unisolated ~/.memtomem/config.json)

## Symptom

On any machine whose real `~/.memtomem/config.json` contains

```json
{ "session_trace": { "enabled": true, "langfuse_enabled": true } }
```

four tests in `packages/memtomem/tests/test_session_tracing.py` fail (`TestConfigConstraints::test_langfuse_validator_keys_missing` plus the three `TestConfigSaveValidationAndRollback` rollback tests), e.g.:

```
>       assert app_mock.config.session_trace.langfuse_enabled is False
E       AssertionError: assert True is False
```

CI never sees this (clean HOME), so it only bites contributors who actually use session tracing. Verified with the same failure on both current `main` (01b5ad28) and 66f251ac — long-standing, not introduced by recent changes.

## Cause

The affected tests construct `Mem2MemConfig()` directly (e.g. `test_session_tracing.py:728`) without isolating the user config file, so the machine's real `~/.memtomem/config.json` leaks into the expected-defaults baseline. Other suites already solve this with the `set_home` helper (`tests/helpers.py`) or an explicit monkeypatched HOME — `test_session_tracing.py:402` even does it for one test — so the fix is mechanical: apply the same isolation to the config-constructing tests in this file.

Sibling of the fresh-HOME class of bugs previously caught in #576-#578 (isolated-HOME smoke).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/packages/memtomem/tests/test_session_tracing.py
+++ b/packages/memtomem/tests/test_session_tracing.py
@@ -723,7 +723,13 @@ class TestConfigSaveValidationAndRollback:
         assert app_mock.state.config.session_trace.langfuse_enabled is False
 
     @pytest.mark.anyio
-    async def test_mcp_config_rollback(self, monkeypatch):
+    async def test_mcp_config_rollback(self, monkeypatch, tmp_path):
+        # Isolate HOME: on a persist failure the MCP rollback rebuilds a fresh
+        # config via load_config_d + load_config_overrides, both of which read
+        # ~/.memtomem (config.d/ and config.json). Without isolation a developer
+        # whose real config.json sets session_trace.langfuse_enabled=true leaks
+        # into the rollback baseline and this assertion flips to True (#1249).
+        set_home(monkeypatch, tmp_path)
         app_mock = MagicMock()
         app_mock.config = Mem2MemConfig()
         app_mock.config.session_trace.enabled = True

